### PR TITLE
Bridging ETH to OP Mainnet With Viem

### PR DESCRIPTION
**Pull request by krofax**
**Description**

This tutorial explains how you can use the Viem to bridge ETH from L1 (Ethereum or Sepolia) to L2 (OP Mainnet or OP Sepolia), it deprecates the Optimism SDK.

**Tests**

Please describe any tests you've added. If you've added no tests, or left important behavior untested, please explain why not.

**Additional context**

Add any other context about the problem you're solving.

**Metadata**

- Fixes https://github.com/ethereum-optimism/devrel/issues/260

### CODE REVIEW

**Review by brokewhale — COMMENTED** [review]
Honestly, I think there might be a bit of a flow break in this tutorial  when viewed from an absolute beginner's perspective. It'd be really helpful if we could make the steps as straightforward and easy to follow as possible.

The full code snippet approach is awesome. If we can have something like the CLI approach, along with a full code snippet, that would be fantastic.

File: pages/builders/app-developers/tutorials/cross-dom-bridge-eth.mdx [hunk 1]
```diff
@@ -1,287 +1,568 @@
...
-
 ## Set Session Variables
 
 You'll need a few variables throughout this tutorial.
 Let's set those up now.
 
-<Steps>
-
-{<h3>Load your private key</h3>}
+<Tabs items={['Create an account', 'Create RPC providers', 'Wallet Clients']}>
```
[the remark this answers is not in the document]
Comment (krofax) [reply pages/builders/app-developers/tutorials/cross-dom-bridge-eth.mdx:113]: @brokewhale i thought about this, but then, inside some tabs are some steps, and we cannot have nested steps.

File: pages/builders/app-developers/tutorials/cross-dom-bridge-eth.mdx [hunk 2]
```diff
@@ -1,287 +1,568 @@
...
+You need to import some dependencies into your project.
 
-<Steps>
+Using `viem`, we will create an account using the private key and initialize L1 and L2 clients to interact with Sepolia and optimismSepolia.
 
-{<h3>Import the Optimism SDK</h3>}
+```js
 
-```js file=<rootDir>/public/tutorials/cross-dom-bridge-eth.js#L3 hash=26b2fdb17dd6c8326a54ec51f0769528
+// Import required modules
```
[the remark this answers is not in the document]
Comment (krofax) [reply pages/builders/app-developers/tutorials/cross-dom-bridge-eth.mdx:98]: Added a step for clarity

**Comment by krofax** [comment]
> Honestly, I think there might be a bit of a flow break in this tutorial when viewed from an absolute beginner's perspective. It'd be really helpful if we could make the steps as straightforward and easy to follow as possible.
> 
> The full code snippet approach is awesome. If we can have something like the CLI approach, along with a full code snippet, that would be fantastic.

What do you mean by a CLI approach?

**Comment by krofax** [comment]
@brokewhale You can check out this tutorial now.

**Comment by brokewhale** [comment]
> @brokewhale You can check out this tutorial now.

on it

**Review by brokewhale — COMMENTED** [review]
some reviews for now. Deposit reviewed 
some things you should note

- some codeblocks are broken
- some are empty
- on the withdrawal part you might want to rename some of the const to be different from deposit to allow continuity 


example
on the deposit side
```js
const args = await publicClientL2.buildDepositTransaction({
    mint: parseEther("0.0001"),
    to: account.address,
});
```

you can't use `args` anymore on the withdrawal side unless you want to restart the node

this will error

```js
//Add the same imports used in DepositETH function
const args = await publicClientL2.buildWithdrawalTransaction({
withdrawalAmount: parseEther("0.0001"),
});
```

instead you can do
```js
const depositArgs = await publicClientL2.buildDepositTransaction({
    mint: parseEther("0.0001"),
    to: account.address,
});
```

and 

```js
//Add the same imports used in DepositETH function
const withdrawalArgs  = await publicClientL2.buildWithdrawalTransaction({
withdrawalAmount: parseEther("0.0001"),
});

and for others too

if the intent is to close the current session and start `node` again while importing all over again then it's fine, but i feel just continuing is better

File: pages/builders/app-developers/tutorials/cross-dom-bridge-eth.mdx [hunk 3]
```diff
@@ -90,198 +78,230 @@ export TUTORIAL_PRIVATE_KEY=0x...
...
 
-```js file=<rootDir>/public/tutorials/cross-dom-bridge-eth.js#L31 hash=3be1dd9a6019e9f59ad6fab38e168e99
-```
+      {<h3>Wait for the L2 transaction to be processed</h3>}
 
-{<h3>Check your wallet balance on L2</h3>}
+      Wait for the L2 transaction to be processed and confirmed and logs the L2 receipt to verify completion.
 
-You should now have more ETH on L2.
+      ```js file=<rootDir>/public/tutorials/cross-dom-bridge-eth.js#L58-L63 hash=357eede3fe4b79bbd160892cfee5d8ec
```
Comment (brokewhale) [inline pages/builders/app-developers/tutorials/cross-dom-bridge-eth.mdx:186]: you want to stop at L62,

there is a trailing `}`

File: public/tutorials/cross-dom-bridge-eth.js [hunk 4]
```diff
@@ -1,74 +1,115 @@
...
+
+async function withdrawETH() {
+    
+//Add the same imports used in DepositETH function
+const withdrawalArgs = await publicClientL2.buildWithdrawalTransaction({
+withdrawalAmount: parseEther("0.0001"),
+to: account.address,
+});
+
+const hash = await walletClientL2.initiateWithdrawal(withdrawalArgs);
```
Comment (brokewhale) [inline public/tutorials/cross-dom-bridge-eth.js:74]:
```js
const withdrawalHash = await walletClientL2.initiateWithdrawal(withdrawalArgs);
```

File: public/tutorials/cross-dom-bridge-eth.js [hunk 5]
```diff
@@ -1,74 +1,115 @@
...
+console.log(`Withdrawal transaction hash on L2: ${hash}`);
```
Comment (brokewhale) [inline public/tutorials/cross-dom-bridge-eth.js:75]: `console.log(`Withdrawal transaction hash on L2: ${withdrawalHash}`);
`

File: public/tutorials/cross-dom-bridge-eth.js [hunk 6]
```diff
@@ -1,74 +1,115 @@
...
 
-console.log('Relaying withdrawal...')
-await messenger.finalizeMessage(withdrawal.hash)
+const receipt = await publicClientL2.waitForTransactionReceipt({ hash });
```
Comment (brokewhale) [inline public/tutorials/cross-dom-bridge-eth.js:77]: `const withdrawalReceipt = await publicClientL2.waitForTransactionReceipt({ withdrawalHash });
`

File: public/tutorials/cross-dom-bridge-eth.js [hunk 7]
```diff
@@ -1,74 +1,115 @@
...
+console.log('L2 transaction confirmed:', receipt);
 
-console.log('Waiting for withdrawal to be relayed...')
-await messenger.waitForMessageStatus(withdrawal.hash, optimism.MessageStatus.RELAYED)
+const { output, withdrawal } = await publicClientL1.waitToProve({
+receipt,
```
Comment (brokewhale) [inline public/tutorials/cross-dom-bridge-eth.js:81]: `withdrawalReceipt `

File: public/tutorials/cross-dom-bridge-eth.js [hunk 8]
```diff
@@ -1,74 +1,115 @@
...
+targetChain: walletClientL2.chain,
+withdrawal,
+});
+
+const finalizeReceipt = await publicClientL1.waitForTransactionReceipt({
+hash: finalizeHash
+});
+
+const status = await publicClientL1.getWithdrawalStatus({
+receipt,
```
Comment (brokewhale) [inline public/tutorials/cross-dom-bridge-eth.js:109]: `withdrawalReceipt `

File: pages/builders/app-developers/tutorials/cross-dom-bridge-eth.mdx [hunk 9]
```diff
@@ -90,198 +78,230 @@ export TUTORIAL_PRIVATE_KEY=0x...
...
-{<h3>Check your wallet balance on L1</h3>}
+      ```js file=<rootDir>/public/tutorials/cross-dom-bridge-eth.js#L104-L106 hash=b3e63b66595ba34594c366e69a451abe
+      ```
 
-You should now have more ETH on L1.
+      {<h3>Check the withdrawal status</h3>}
 
-```js file=<rootDir>/public/tutorials/cross-dom-bridge-eth.js#L72 hash=3be1dd9a6019e9f59ad6fab38e168e99
-```
+      ```js file=<rootDir>/public/tutorials/cross-dom-bridge-eth.js#L108-L113 hash=d23afa6efca5573a71a1fe8e1205f997
```
Comment (brokewhale) [inline pages/builders/app-developers/tutorials/cross-dom-bridge-eth.mdx:278]: should end at L112

File: public/tutorials/cross-dom-bridge-eth.js [hunk 10]
```diff
@@ -1,74 +1,115 @@
...
-    throw new Error('Timed out waiting for block to be produced')
-  }
-  await new Promise(resolve => setTimeout(resolve, 1000))
-}
+const depositArgs = await publicClientL2.buildDepositTransaction({
+    mint: parseEther("0.0001"),
+    to: account.address,
+});
+
+const Deposithash = await walletClientL1.depositTransaction(depositArgs);
```
Comment (brokewhale) [inline public/tutorials/cross-dom-bridge-eth.js:49]: depositHash

**Review by brokewhale — APPROVED** [review]

File: public/tutorials/cross-dom-bridge-eth.js [hunk 11]
```diff
@@ -1,74 +1,115 @@
...
+} 
+
+
+async function withdrawETH() {
+    
+//Add the same imports used in DepositETH function
+const withdrawalArgs = await publicClientL2.buildWithdrawalTransaction({
+withdrawalAmount: parseEther("0.0001"),
+to: account.address,
+});
```
Comment (brokewhale) [inline public/tutorials/cross-dom-bridge-eth.js:72]:
```js

// Build parameters to initiate the withdrawal transaction on the L1.
const withdrawalArgs = await publicClientL1.buildInitiateWithdrawal({
  to: account.address,
  value: parseEther('0.0001')
})
 ```

**Review by brokewhale — CHANGES_REQUESTED** [review]
all good except this

**Comment by krofax** [comment]
> all good except this

Done, please approve.

**Review by brokewhale — APPROVED** [review]

File: pages/builders/app-developers/tutorials/cross-dom-bridge-eth.mdx [hunk 12]
```diff
@@ -1,30 +1,25 @@
 ---
-title: Bridging ETH with the Optimism SDK
+title: Bridging ETH with Viem
```
Comment (bradleycamacho) [inline pages/builders/app-developers/tutorials/cross-dom-bridge-eth.mdx:2]:
```suggestion
title: Bridging ETH with viem
```

File: pages/builders/app-developers/tutorials/cross-dom-bridge-eth.mdx [hunk 13]
```diff
@@ -1,30 +1,25 @@
...
-description: Learn how to use the Optimism SDK to transfer ETH between Layer 1 (Ethereum or Sepolia) and Layer 2 (OP Mainnet or OP Sepolia).
+description: Learn how to use Viem to transfer ETH between Layer 1 (Ethereum or Sepolia) and Layer 2 (OP Mainnet or OP Sepolia).
 ---
 
-import { Callout, Steps } from 'nextra/components'
-import { WipCallout } from '@/components/WipCallout'
+import { Callout, Steps, Tabs } from 'nextra/components'
 
-<WipCallout />
+# Bridging ETH to OP Mainnet With Viem
```
Comment (bradleycamacho) [inline pages/builders/app-developers/tutorials/cross-dom-bridge-eth.mdx:9]:
```suggestion
# Bridging ETH to OP Mainnet with viem
```

File: pages/builders/app-developers/tutorials/cross-dom-bridge-eth.mdx (hunk 12 above)
Comment (bradleycamacho) [reply pages/builders/app-developers/tutorials/cross-dom-bridge-eth.mdx:2]: Their docs are really sketchy in regards to accurate use, but it seems that `viem` should always be lowercase. I'll only mark one or two instances, but you should take a pass over the entire PR and lowercase it

https://viem.sh/docs/introduction

File: pages/builders/app-developers/tutorials/cross-dom-bridge-eth.mdx [hunk 14]
```diff
@@ -90,198 +78,230 @@ export TUTORIAL_PRIVATE_KEY=0x...
...
+  ```js file=<rootDir>/public/tutorials/cross-dom-bridge-eth.js#L68-L111 hash=cd8d5c4858141d43b519cd7267cd89a9
+  ```
+
+  </Tabs.Tab>
+</Tabs>
+
+## Important Considerations
+
+<Callout type="warning">
+  *   Challenge Period: The 7-day withdrawal challenge Period is crucial for security.
```
Comment (bradleycamacho) [inline pages/builders/app-developers/tutorials/cross-dom-bridge-eth.mdx:294]:
```suggestion
  *   Challenge period: The 7-day withdrawal challenge Period is crucial for security.
```

File: pages/builders/app-developers/tutorials/cross-dom-bridge-eth.mdx [hunk 15]
```diff
@@ -90,198 +78,230 @@ export TUTORIAL_PRIVATE_KEY=0x...
...
+  *   Gas Costs: Withdrawals involve transactions on both L2 and L1, each incurring gas fees.
```
Comment (bradleycamacho) [inline pages/builders/app-developers/tutorials/cross-dom-bridge-eth.mdx:295]:
```suggestion
  *   Gas costs: Withdrawals involve transactions on both L2 and L1, each incurring gas fees.
```

File: pages/builders/app-developers/tutorials/cross-dom-bridge-eth.mdx [hunk 16]
```diff
@@ -90,198 +78,230 @@ export TUTORIAL_PRIVATE_KEY=0x...
...
+  *   Private Key Handling: Use secure key management practices in real applications.
```
Comment (bradleycamacho) [inline pages/builders/app-developers/tutorials/cross-dom-bridge-eth.mdx:296]:
```suggestion
  *   Private Key handling: Use secure key management practices in real applications.
```

Comment (bradleycamacho) [reply pages/builders/app-developers/tutorials/cross-dom-bridge-eth.mdx:296]: Not sure if Key is capitalized

File: pages/builders/app-developers/tutorials/cross-dom-bridge-eth.mdx [hunk 17]
```diff
@@ -90,198 +78,230 @@ export TUTORIAL_PRIVATE_KEY=0x...
...
+  *   RPC Endpoint Security: Keep your API key (or any RPC endpoint) secure.
```
Comment (bradleycamacho) [inline pages/builders/app-developers/tutorials/cross-dom-bridge-eth.mdx:297]:
```suggestion
  *   RPC endpoint security: Keep your API key (or any RPC endpoint) secure.
```

File: pages/builders/app-developers/tutorials/cross-dom-bridge-eth.mdx [hunk 18]
```diff
@@ -90,198 +78,230 @@ export TUTORIAL_PRIVATE_KEY=0x...
...
+</Callout>
 
 ## Next Steps
 
-Congrats!
-You've just deposited and withdrawn ETH using the Optimism SDK.
-You should now be able to write applications that use the Optimism SDK to transfer ETH between L1 and L2.
+*   Develop a user interface for easier interaction with these bridging functions.
```
Comment (bradleycamacho) [inline pages/builders/app-developers/tutorials/cross-dom-bridge-eth.mdx:302]: Do we have any resources we can share here? These are two really large "steps" and it'd be nice to give users a breadcrumb to follow at the least

**Review by bradleycamacho — CHANGES_REQUESTED** [review]
Just a few changes! Nothing large, good job on this!

**Review by bradleycamacho — APPROVED** [review]

**Comment by fujianggara10** [comment]
Good job fellow

**Pull request merged**